src/inspectors/GitInspector.spec.ts

```diff
@@ -67,6 +67,110 @@ describe('GitInspector', () => {
+      await testDir.gitCommit('msg2', 'test2 <test2@example.com>');
+      const expected = (await testDir.gitLog()).latest;
+      const gitInspector = new GitInspector(testDir.path);
+
+      const { items } = await gitInspector.getCommits({ filter: { author: 'test2@example.com' } });
+
+      expect(items).toStrictEqual([
+        {
+          sha: expected.hash,
+          date: new Date(expected.date),
+          message: 'msg2\n',
+          author: { name: 'test2', email: 'test2@example.com' },
+          commiter: undefined,
+        },
+      ]);
+    });
+
+    it('returns repository commits between sha and HEAD', async () => {
+      await testDir.gitInit({ name: 'test', email: 'test@example.com' });
+      await testDir.gitCommit('msg1');
+      const commit1 = (await testDir.gitLog()).latest;
+      await testDir.gitCommit('msg2');
+      const commit2 = (await testDir.gitLog()).latest;
+      const gitInspector = new GitInspector(testDir.path);
+
+      const { items } = await gitInspector.getCommits({ filter: { sha: commit1.hash } });
+
+      expect(items).toStrictEqual([
+        {
+          sha: commit2.hash,
+          date: new Date(commit2.date),
+          message: 'msg2\n',
+          author: { name: 'test', email: 'test@example.com' },
+          commiter: undefined,
+        },
+      ]);
+    });
+
+    it('returns repository commits containing a path', async () => {
+      await testDir.gitInit({ name: 'test', email: 'test@example.com' });
+      await testDir.gitCommit('msg1');
+      await testDir.gitAddFile('file.txt');
+      await testDir.gitCommit('msg2');
+      const expected = (await testDir.gitLog()).latest;
+      const gitInspector = new GitInspector(testDir.path);
+
+      const { items } = await gitInspector.getCommits({ filter: { path: 'file.txt' } });
+
+      expect(items).toStrictEqual([
+        {
+          sha: expected.hash,
+          date: new Date(expected.date),
+          message: 'msg2\n',
+          author: { name: 'test', email: 'test@example.com' },
+          commiter: undefined,
+        },
+      ]);
+    });
+
+    it('returns repository commits after a date', async () => {
+      await testDir.gitInit({ name: 'test', email: 'test@example.com' });
+      await testDir.gitCommit('msg1');
+      await new Promise((resolve) => setTimeout(resolve, 1000)); // sleep 1 s
+      await testDir.gitCommit('msg2');
+      const expected = (await testDir.gitLog()).latest;
+      const gitInspector = new GitInspector(testDir.path);
+
+      const { items } = await gitInspector.getCommits({ filter: { since: new Date(expected.date) } });
+
+      expect(items).toStrictEqual([
+        {
+          sha: expected.hash,
+          date: new Date(expected.date),
+          message: 'msg2\n',
+          author: { name: 'test', email: 'test@example.com' },
+          commiter: undefined,
+        },
+      ]);
+    });
+
+    it('returns repository commits before a date', async () => {
```

Comment:
Maybe 'until' could be better wording than 'before'

Reply:
I've borrowed the wording from GitHub docs but doesn't matter... Changed in 
656e4a0.